### PR TITLE
Add reducers and actions to the app

### PR DESCRIPTION
**Pull request by JonahKayizzi**
This pull request consists of the following;

- Added Redux Toolkit.
- Structured the application files using a "feature folder" approach and use the ducks pattern
- Implemented the book's actions and action types for adding and deleting a book and the reducer
- Implemented the categories actions, action types and reducer with a return string "Under construction".
- Configured the Redux Store
- Combined both reducers into a root reducer by using configureStore function.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,7 +3,19 @@ import { Route, Routes } from 'react-router-dom';
...
+import { checkCategoryStatus } from './redux/categories/categories';
 
+store.dispatch(addBook({ id: 1, title: 'The End', author: 'Michael' }));
+store.dispatch(addBook({ id: 2, title: 'Microverse', author: 'Ariel' }));
+console.log(store.getState());
+store.dispatch(deleteBook({ id: 1 }));
+console.log(store.getState());
+store.dispatch(checkCategoryStatus(2));
+console.log(store.getState());
+*/
```
Comment (t-yanick) [inline src/App.js]: - [ ] Please kindly remove commented code from your code base as this makes your  code dirty and less readable.

File: src/redux/types/types.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const BOOK_ADDED = 'BOOK_ADDED';
+const BOOK_DELETED = 'BOOK_DELETED';
+const STATUS_CHECKED = 'STATUS_CHECKED';
```
Comment (t-yanick) [inline src/redux/types/types.js]: - [ ] As it is a [requirement ](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md)for this project, each action type needs to follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux), please kindly fix that.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @JonahKayizzi, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N.A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/types/types.js (hunk 2 above)
Comment (JonahKayizzi) [reply src/redux/types/types.js]: Thanks for the suggestion. It is to be effected

File: src/App.js (hunk 1 above)
Comment (JonahKayizzi) [reply src/App.js]: Thanks for the suggestion. This has been corrected.

**Review by t-yanick — APPROVED** [review]
Hi @JonahKayizzi , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**